File: restapi/services/auth_service/src/myapp/container/container.py
from myapp.config import Config, load_config
from myapp.db import DbHandler
from myapp.db.mdb import MariaDbRepository
from myapp.logger.logger import log


class Container:
    config: Config
    db_hnd: DbHandler


# _container: Container | None = None


def new_container() -> Container:
    global _container

    c = Container()

    # load config
    try:
        c.config = _init_config()
    except Exception as e:
        raise RuntimeError(f"config loading error..{e}") from e

    # init database
    c.db_hnd = _init_database(c.config)

    # _container = c
    return c


def _init_config() -> Config:
    return load_config(".")


def _init_database(cfg: Config) -> DbHandler:
    repo = MariaDbRepository(
        user=cfg.db_user,
        pw=cfg.db_passwd,
        dbname=cfg.db_name,
        host=cfg.db_address,
        port=cfg.db_port,
    )
    try:
        repo.init()
    except Exception as e:
        log.error("Db Init err.. {}", str(e))
    return repo

File: restapi/services/auth_service/ref/container/container.go
package container

import (
	"auth-service/internal/config"
	"auth-service/internal/db"
	"auth-service/internal/db/mdb"
	"auth-service/internal/logger"
	"fmt"
)

type Container struct {
	Config *config.Config
	DbHnd  db.DbHandler
}

var container *Container

func NewContainer() (*Container, error) {
	container = &Container{}
	// load config
	config, err := initConfig()
	if err != nil {
		return nil, fmt.Errorf("config loading error..%v \n", err)
	}
	container.Config = &config

	// init database
	dbhnd := initDatabase(config)
	container.DbHnd = dbhnd

	return container, nil
}

func initConfig() (config.Config, error) {
	return config.LoadConfig(".")
}

func initDatabase(config config.Config) db.DbHandler {
	mdb := mdb.NewMdbHandler(config.DBUser, config.DBPasswd, config.DBSName, config.DBAddress, config.DBPort)
	err := mdb.Init()
	if err != nil {
		logger.Log.Error("Db Init err.. %v", err)
	}
	return mdb
}
